Document sage installation method with pip

zovelsanj opened the pull request:
<!-- ^^^^^
Please provide a concise, informative and self-explanatory title.
Don't put issue numbers in there, do this in the PR body below.
For example, instead of "Fixes #1234" use "Introduce new method to calculate 1+1"
-->
### 📚 Description

<!-- Describe your changes here in detail -->
<!-- Why is this change required? What problem does it solve? -->
<!-- If it resolves an open issue, please link to the issue here. For example "Closes #1337" -->

### 📝 Checklist

<!-- Put an `x` in all the boxes that apply. -->
<!-- If your change requires a documentation PR, please link it appropriately -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] I have made sure that the title is self-explanatory and the description concisely explains the PR.
- [x] I have linked an issue or discussion.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation accordingly.

### ⌛ Dependencies
<!-- List all open pull requests that this PR logically depends on -->
<!--
- #xyz: short description why this is a dependency
- #abc: ...
-->

zovelsanj committed:
updated sage installation method with pip

Added sage library installation instructions using pip. Installing sage in a python virtual environment can be troublesome as `sage` and `sagemath` libraries, not at all related to this sage, exist in PiPy.

zovelsanj committed:
Merge pull request #1 from zovelsanj/zovelsanj-patch-1

updated sage installation method with pip

mkoeppe commented on README.md:
```diff
@@ -394,6 +394,18 @@ in the Installation Guide.
     in the installation manual.
+    
+Alternative Installation using PiPy
+---------------
```
check spelling

mkoeppe commented on README.md:
```diff
@@ -394,6 +394,18 @@ in the Installation Guide.
+Alternative Installation using PiPy
+---------------
+
+You can find `sage` and `sagemath` pip packages but it is worth nothing that they are not at all realted to this `sage`. Indeed `pip install sage` installs `sage 0.0.0` which is literally an empty package. For installation of this `sage` you need to install `sagemath-standard`. First activate your python virtual environment and follow these steps:
```
I don't think it is needed to talk about these packages.

zovelsanj replied:
These packages are very misleading during sage installation in a python virtual environment and the sources on the internet do not explain clearly about the installation of sage package using pip.

mkoeppe replied:
At least don't start the writing with what shouldn't be done. And nobody needs to know in what specific way these packages are unsuitable. As simple "(Ignore the `sage` and `sagemath` packages on PyPI.)" should be enough.

zovelsanj replied:
Agreed. Thank you

zovelsanj replied:
Perhaps a note would be better to avoid this ambiguity of `sage` and `sagemath` in PyPi.

alexjbest replied:
Perhaps we should contact the maintainer of the empty sage package, @dhagrow, and ask if they would mind transferring the name to the SageMath org somehow (unsure if this ping here does anything but it's fairly easy to find an associated email address for that user on google if we did want to get in conact if desired).

mkoeppe replied:
That's #28796.

zovelsanj committed:
fixed spelling errors

zovelsanj committed:
pip packages instructions added in note 

Fixed the pip installation instructions according to this [comment](https://github.com/sagemath/sage/pull/35070#discussion_r1103552497).

mkoeppe commented on README.md:
```diff
@@ -394,6 +394,20 @@ in the Installation Guide.
     "Launching SageMath"](https://doc.sagemath.org/html/en/installation/launching.html)
     in the installation manual.
+    
+Alternative Installation using PiPy
```
It's "PyPI"

zovelsanj replied:
sorry, a typo. Fixed now.

zovelsanj committed:
fixed typos

changed `PiPy` to 'PyPi'

mkoeppe commented on README.md:
```diff
@@ -394,6 +394,20 @@ in the Installation Guide.
     in the installation manual.
+    
+Alternative Installation using PyPi
+---------------
```
It's PyPI with a capital I

zovelsanj replied:
sorry. Corrected now.

mkoeppe commented on README.md:
```diff
@@ -394,6 +394,20 @@ in the Installation Guide.
+
+You need to install the `sage_conf`, a wheelhouse of various python packages. You can list the wheels using `ls $(sage-config SAGE_SPKG_WHEELS)`. After manual installation of these wheels, you can install the sage library, `sagemath-standard`. 
+
+**NOTE:** You can find `sage` and `sagemath` pip packages but it is worth noting that they are not at all related to this `sage`. Indeed `pip install sage` installs `sage 0.0.0` which is literally an empty package. 
```
These packages _are_ actually related to SageMath, but they are not useful. 
Please reduce

zovelsanj replied:
After installing `sage` and `sagemath` with `pip` and importing `sage` it throws an import error: `No module named sage`. 
I don't understand how they are related to sage if they can't import sage at all. Please refer to [this](https://stackoverflow.com/questions/50039916/no-module-named-sage-all) post on stackoverflow.

mkoeppe replied:
There's nothing deep to understand here. Both packages were created because the respective authors intended to provide some functionality related to SageMath. But neither of the two packages can be used to install SageMath, so they should be ignored.

zovelsanj replied:
Ok, I will update the readme

zovelsanj committed:
fixed a typo

changed PyPi tp PyPI

alexjbest commented on README.md:
```diff
@@ -394,6 +394,20 @@ in the Installation Guide.
+Alternative Installation using PyPI
+---------------
+
+For installation of `sage` in python using `pip` you need to install `sagemath-standard`. First, activate your python virtual environment and follow these steps:
```
```suggestion
For installation of `sage` in python using `pip` you need to install `sagemath-standard`. First, activate your python virtual environment and follow these steps:
```

alexjbest commented on README.md:
```diff
@@ -394,6 +394,20 @@ in the Installation Guide.
+            $ python3 -m pip install $(sage-config SAGE_SPKG_WHEELS)/*.whl 
+            $ python3 -m pip install sagemath-standard
+
+You need to install the `sage_conf`, a wheelhouse of various python packages. You can list the wheels using `ls $(sage-config SAGE_SPKG_WHEELS)`. After manual installation of these wheels, you can install the sage library, `sagemath-standard`. 
```
```suggestion
You need to install `sage_conf`, a wheelhouse of various python packages. You can list the wheels using `ls $(sage-config SAGE_SPKG_WHEELS)`. After manual installation of these wheels, you can install the sage library, `sagemath-standard`. 
```

zovelsanj committed:
updated sage pip note

updated the ambiguity note on `sage` and `sagemath` as per [#35070](https://github.com/sagemath/sage/pull/35070#discussion_r1103746908)

mkoeppe commented on README.md:
```diff
@@ -407,7 +407,7 @@ For installation of `sage` in python using `pip` you need to install `sagemath-s
 You need to install the `sage_conf`, a wheelhouse of various python packages. You can list the wheels using `ls $(sage-config SAGE_SPKG_WHEELS)`. After manual installation of these wheels, you can install the sage library, `sagemath-standard`. 
 
-**NOTE:** You can find `sage` and `sagemath` pip packages but it is worth noting that they are not at all related to this `sage`. Indeed `pip install sage` installs `sage 0.0.0` which is literally an empty package. 
+**NOTE:** You can find `sage` and `sagemath` pip packages but it is worth noting that when importing `sage` with these packages you will encounter `ModuleNotFoundError: No module named 'sage'`. Indeed `pip install sage` installs `sage 0.0.0` which is literally an empty package. 
```
Still too much information.

zovelsanj replied:
reduced the information.

zovelsanj committed:
Merge branch 'sagemath:develop' into develop

zovelsanj committed:
reduced information

Reduced `sage` and `sagemath` information as per [#35070](https://github.com/sagemath/sage/pull/35070#discussion_r1103826628)

Commit:
README.md: Fix wording

Co-authored-by: Alex J Best <alex.j.best@gmail.com>

mkoeppe reviewed (approved):
LGTM now, thanks!
